WIP: Separate credentialing and training

fiedosiukr opened the pull request:
## Changes
- added separate tab in the user settings page - Trainings
- trainings and credentialing are separate processes now
- there are different training types with custom valid duration
- reviewer questions are customizable for each training type

## To do
- add a mechanism to allow project authors to select trainings that are required to access the data set

Commit:
Add Training models.

Commit:
Add TrainingForm.

Commit:
Add Training section to the settings page.

Commit:
Remove training data from the fixures.

Commit:
Add Trainings section to the admin console.

Commit:
Remove training part from the credentialing.

Commit:
Add conditional rendering.

Commit:
Format the code.

Commit:
Add conditional access form field rendering.

Commit:
Modify access form in the copyedit phase.

Commit:
Improve trainings.

Commit:
Add test cases.

Commit:
Modify demo-files.

Commit:
Improve training processing.

alistairewj commented on physionet-django/user/templates/user/edit_trainings.html:
```diff
@@ -0,0 +1,57 @@
+{% extends "user/settings.html" %}
+
+{% block title %}PhysioNet Trainings{% endblock %}
```
`{{ SITE_NAME }} Training`?

fiedosiukr replied:
Nice catch! It's been rebased after the debranding PR was merged and I forgot about that.

alistairewj commented on physionet-django/user/templates/user/edit_trainings.html:
```diff
@@ -0,0 +1,57 @@
+    {% endfor %}
+  </ul>
+  <hr>
+  <h4>New training</h4>
```
Submit evidence of a completed training course.

alistairewj commented on physionet-django/user/templates/user/edit_trainings.html:
```diff
@@ -0,0 +1,57 @@
+  </ul>
+  <hr>
+  <h4>New training</h4>
+  {% lorem 1 p %}
```
Provide a completion report indicating that you have finished the respective training course.

alistairewj commented on physionet-django/user/templates/user/edit_trainings.html:
```diff
@@ -0,0 +1,57 @@
+{% block main_content %}
+<h1>PhysioNet Trainings</h1>
+  <hr>
+  <h4>Active trainings</h4>
```
try to replace "trainings" with "training" wherever you see it

Commit:
Change trainings to training.

Commit:
Fix migrations.

Commit:
Format the code.

Commit:
Add required training to contributor review access policy.

Commit:
Fix migrations.

Commit:
Add missing demo files.

Commit:
Replace placeholder text placeholders.

Commit:
Improve file access section.

tompollard committed:
display 'contact reference' button

tompollard committed:
display applications at 'final approval' stage

tompollard committed:
rm comment

tompollard committed:
Add an 'accept all' button for training review decisions.

tompollard committed:
add notification email for training applications.

send an email to the applicant when their training application has been processed.

tompollard committed:
Remove comments about training

Training is no longer checked during the credentialing process, so comments about training are not relevant.

tompollard committed:
Highlight training requirement in credentialing decision email.

Training has been separated from credentialing. If credentialing is approved, we need to remind people that training may also be necessary.

tompollard committed:
Update email that is sent to users when they apply for credentialed access.

Currently we send an email to users when they apply for credentialed access. The email doesn't make a great deal of sense because it focuses on DUAs, which are now handled on a dataset level.

tompollard committed:
add user profile to tests.

tompollard committed:
fix formatting.

Commit:
Fix missing characters in root-level directories in GCP

Commit:
Fix empty text files being created when copying GCS folders

Commit:
Fix uploading root-level files in GCS

Commit:
Fix individual_size_limit missing after dir navigation

adrianryt committed:
Display training status in user management page

adrianryt committed:
Mention trainings on credentialing page

Commit:
Separate trainings by type in the user settings view

Commit:
Fix flake8 warnings

adrianryt committed:
Add filler for introductory sentence and CITI link

Commit:
Merge remote-tracking branch 'origin/dev' into feature/separate-training-and-credentialing

Commit:
Merge branch 'dev' into feature/separate-training-and-credentialing

tompollard committed:
update wording on training and credentialing. change trainings to training throughout

update the wording on training and credentialing for clarity. also change 'trainings' to 'training'.
